packages/build/types/index.d.ts

```diff
@@ -1 +1,6 @@
 export { NetlifyPlugin } from './netlify_plugin'
+export { NetlifyPluginOptions } from './netlify_plugin_options'
+export { NetlifyEventHandler } from './netlify_event_handler'
```

Comment:
If we're willing to export this - maybe it's better to export the specific hook functions? i.e. `type OnPreBuild<...>`, `type OnPostBuild<...>` etc., instead of the generic `NetlifyEventHandler` - which is an implementation detail more than anything.

I wrote about this already in [a previous comment](https://github.com/netlify/build/pull/3778#issuecomment-958049423) (or more?), and it should be implemented in a separate PR by @ehmicky AFAIK. Maybe we can integrate it into this one - if you want to take a shot at it @ascorbic?

Reply:
They all have the same signature, which is why I didn't bother.

Reply:
Based on the fact that event handlers have slightly different signatures (the `error` parameter) at the moment, and might diverge further in the future, it does make sense to me to export each event handler type separately, instead of their "base" type.

What are your thoughts on this @ascorbic?

Reply:
Yes, I'm fine with that